### ISSUE TITLE
Removal of Metasploit from the Microsoft Defender Database

### ISSUE THREAD

**Issue by AlexDev404** [rapid7/metasploit-framework#16636]
## Summary
Removal of Metasploit from the Microsoft Defender database.

## Relevant information
Hi,

I'd like to ask a favor of you all.

I am trying to add Metasploit to the Microsoft[ Winget Package Database](https://github.com/microsoft/winget-pkgs/) but am having trouble since Microsoft has detected it as [malware](https://github.com/microsoft/winget-pkgs/pull/61780#issuecomment-1135208321).

What I'd like to ask from you all is if you can, when you get the chance to please submit a [dispute](https://www.microsoft.com/en-us/wdsi/filesubmission?persona=SoftwareDeveloper) to Microsoft so that the false-positive can be removed from the database? It would really help all of us, including me.

> It seems as though I cannot file this dispute as an individual, but rather, the dispute must
> be filled out by **the developer(s)** who created Metasploit, which is why I am contacting you all.

Here is the link to the dispute form:
https://www.microsoft.com/en-us/wdsi/filesubmission?persona=SoftwareDeveloper

And this is the file I used in the [pull request](https://github.com/microsoft/winget-pkgs/pull/61780/files#diff-e0d00e1dfc789da53623d61f8b77d41f013176a6258a4e20f3abf15de3d9eafbR10):
https://windows.metasploit.com/metasploitframework-latest.msi

Thanks in advance!

**Comment by smcintyre-r7** [comment]
TLDR; I'm sorry but I don't think we'll be pursuing this.

The thing is that if we hypothetically succeeded in this endeavor and got the MSI installer removed from the list so a user could install Metasploit, the installation would likely be corrupt. We do attempt to [detect this](https://github.com/rapid7/metasploit-framework/blob/47fcf541e332e8be7a71ce41738be712a29071ee/lib/msf/core/framework.rb#L240-L257) ourselves but that's more of a fail-safe and is itself prone to errors due to all the variations in how the different AV products behave. We also call this out as a known issue in [the wiki](https://docs.metasploit.com/docs/using-metasploit/getting-started/nightly-installers.html#windows-anti-virus-software-flags-the-contents-of-these-packages) and suggest disabling [anti-virus](https://docs.metasploit.com/docs/using-metasploit/getting-started/reporting-a-bug.html#other-notes). The reason the installation would likely be corrupted is that we have a lot of exploits (see the contents of [`data/exploits`](https://github.com/rapid7/metasploit-framework/tree/master/data/exploits)) and payloads in their binary forms that have the necessary components patched. These common data files are a sort of bottleneck in terms of evasion that are often thoroughly signatured and for good reason. So in order to truly allow-list a Metasploit installation these signatures would also need to be updated or removed which would likely do much more harm than good.

If you'd like to install Metasploit, I suggest you handle the necessary Anti-Virus reconfiguration prior to starting the process. Knowing the desired installation directory is a good first step.

**Comment by Denalang** [comment]
It would have been nice if you lot could have made that clear from the start, I trusted your download and now have corrupted my laptop with a virus I cannot remove that has caused untold problems with my system.
Den
